cranelift/fuzz/fuzz_targets/fuzzgen.rs

```diff
@@ -0,0 +1,90 @@
+#![no_main]
+
+use libfuzzer_sys::fuzz_target;
+
+use arbitrary::Unstructured;
+use cranelift::codegen::data_value::DataValue;
+use cranelift::codegen::ir::Function;
+use cranelift::prelude::*;
+use cranelift_filetests::function_runner::{CompiledFunction, SingleFunctionCompiler};
+use cranelift_fuzzgen::*;
+use cranelift_interpreter::environment::FuncIndex;
+use cranelift_interpreter::environment::FunctionStore;
+use cranelift_interpreter::interpreter::{Interpreter, InterpreterState};
+use cranelift_interpreter::step::ControlFlow;
+use cranelift_interpreter::step::CraneliftTrap;
+
+enum RunResult {
+    Success(Vec<DataValue>),
+    Trap(CraneliftTrap),
+    Error(Box<dyn std::error::Error>),
+}
+
+fn run_in_interpreter(interpreter: &mut Interpreter, args: &[DataValue]) -> RunResult {
+    // The entrypoint function is always 0
+    let index = FuncIndex::from_u32(0);
+    let res = interpreter.call_by_index(index, args);
```

Comment:
> when fuzz-case generation becomes sophisticated enough for that, if it's not already?

It isn't simply because we limit the number of generated instructions up to a maximum of 16. Otherwise I think we are simply limited by the input length. On a related note, I moved all of the arbitrary restrictions that we place on the generator into a `Config` module so that this become more obvious (and so that we can eventually do swarm fuzzing).

> There's a question whether we step for N steps on both sides (interp and compiled) and then compare state at the end, or if we just reject inputs that run that long. I suspect probably the latter is good enough

Rejecting inputs that run that long on the interpreter seems like the way to go for me.